test/runtests.jl

```diff
@@ -1,6 +1,22 @@
 using Expokit
 using Base.Test
 
+struct LinearOp
+    m
+end
```

Comment:
wouldn't this be better as:
```
struct LinearOp{T}
    m::T
end
```

Reply:
You're right, thanks for pointing this out, add a type parameter can increase type stability a lot.
Since this `LinearOp` is only used for testing and this branch has been merged, I probability won't fix it right now. User are suggested to use [LinearMaps.jl](https://github.com/Jutho/LinearMaps.jl) package, `LinearMap` type in this package is much more powerful and supports many abstract operations.

Reply:
Thanks.  Let me explain a bit more why I asked.  Since there is no documentation in this pull request, I looked at the code because I was a bit confused how to use this functionality.  Although the pull request mentions LinearMaps, I actually mistakenly started to think that support for LinearMaps had been removed during the review process of this pull request in favor of `LinearOp`.

That said, why not add `LinearMaps` to `test/REQUIRE` and use it in the tests if the goal is to support it?

Reply:
I think the title of the PR was a bit misleading. Basically, a new keyword `anorm`was added which also lifts the need for `A`to support norm/opnorm and hence allows using `LinearMaps`. As @GiggleLiu says the test is not really an example for using `LinearMaps`. 

Maybe it would be a good a idea to add such an example to README or somewhere else, though?

Reply:
@GiggleLiu Sure, it is on the list. OTOH there isn't that much to document ...